File: 初级算法/对称二叉树/C++/original.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 * 思路和python一样
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        if(root==NULL) return true;
        return ValidBST(root->left, root->right);
    }
    bool ValidBST(TreeNode* L, TreeNode* R){
        if(L==NULL) return R== NULL;
        if(R==NULL) return L==NULL;
        if(R->val==L->val) return ValidBST(L->left, R->right) && ValidBST(L->right, R->left);
        else return false;
    }
};

File: 初级算法/二叉树的最大深度/python3/original.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        思路：和C++一样
        """
        return root and max(self.maxDepth(root.left), self.maxDepth(root.right))+1 or 0